src/renderers/shared/fiber/ReactChildFiber.js

```diff
@@ -1100,6 +1101,41 @@ function ChildReconciler(shouldClone, shouldTrackSideEffects) {
     // not as a fragment. Nested arrays on the other hand will be treated as
     // fragment nodes. Recursion happens at the normal flow.
 
+    if (ReactFeatureFlags.disableNewFiberFeatures) {
+      // Support only the subset of return types that Stack supports. Treat
+      // everything else as empty, but log a warning.
+      if (typeof newChild === 'object' && newChild !== null) {
+        switch (newChild.$$typeof) {
+          case REACT_ELEMENT_TYPE:
+            return placeSingleChild(reconcileSingleElement(
+              returnFiber,
+              currentFirstChild,
+              newChild,
+              priority
+            ));
+
+          case REACT_PORTAL_TYPE:
+            return placeSingleChild(reconcileSinglePortal(
+              returnFiber,
+              currentFirstChild,
+              newChild,
+              priority
+            ));
+        }
+      }
+
+      if (__DEV__) {
+        const Component = returnFiber.type;
+        warning(
```

Comment:
>Occasionally we do that intentionally too if the warning gives a better stack.

Yep, but isn’t the case there. (Both stacks are internal and roughly in the same place.)

>I'll rebase on top

You probably only want da3dc0354dc37af431f59ac04a29e694515511cd, 243d24542354eae55a77ce68ea556e04a13967be, 2646272e8ac55a15bbfde0dc49b0e5294b7cd5fd.

I haven't looked at your PR yet but you might want to do 8c4b2a64dbb848bd6659821b0c3af771f283e03c and 5643a2c6272ca27cff5c4b739638cce266ee1dc6 differently.